src/vocabulary-cards/src/app/services/history.service.ts

```diff
@@ -0,0 +1,32 @@
+import {Injectable} from '@angular/core';
+import {StorageService} from "./storage.service";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class HistoryService {
+    constructor(private storage: StorageService) {
+    }
+
+    addDateWithTraining(timestamp: number) {
+        let learned: Set<number> | null = this.storage.get<Set<number>>("learned");
+        learned?.add(timestamp);
+        this.storage.set<Set<number>>("learned", learned!);
```

Comment:
This code is incorrect.
If the storage returns null, the `learned?.add(timestamp);` will not do anything with it and you will just set null back to the storage.

You have the `getDatesWithTraining()` method in the HistoryService class. Use it. Get a Set from that method, add a new value into it and save it to the storage.